internal/telemetry/telemetry.go

```diff
@@ -7,17 +7,50 @@
 // Datadog regarding usage of an APM library such as tracing or profiling.
 package telemetry
 
-// ProductChange enqueues an app-product-change event that signals a product has been turned on/off.
+import "gopkg.in/DataDog/dd-trace-go.v1/internal/appsec"
+
+// ProductStart signals that the tracer or profiler has started with
+// some configuration information. The telemetry event emitted depends
+// on which product is starting, as well as whether an app-started event
+// has already been sent.
+func (c *client) ProductStart(namespace Namespace, configuration []Configuration) {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+	if c.started {
+		switch namespace {
```

Comment:
There should be a `default` where we do something (or log something).

Also, why can't `NamespaceASM` be a valid input to this function? Even if it's never called today, shouldn't it be for future proofing?

Reply:
Just added a default case with a log. For the ASM case, that makes sense, thanks for the suggestion - i'll add a case in `select namespace` to catch ASM, so that in the future, ASM has the option to directly access the telemetry client.